modules/packages/Sort.chpl

```diff
@@ -18,27 +18,178 @@
  */
 
 /*
-   The `Sort` module is designed to support standard sorting
-   routines.  The current interface is minimal and should be expected to
-   grow and evolve over time. Each of the following functions accepts an
-   optional boolean argument, `reverse`, which is `false` by
-   default. If `true`, then the sort function will order in reverse.
+
+This module supports a variety of standard sorting routines on 1D arrays.
+
+The current interface is minimal and should be expected to
+grow and evolve over time.
+
+Comparators
+-----------
+
+Comparators allow sorting data by a mechanism other than the
+default comparison of array elements. To use a comparator, define a
+record with either a ``key(a)`` or ``compare(a, b)`` method, and pass an
+instance of that record to the sort function.
+
+Most sort routines support comparators, which is denoted by their function
+signature.
+
+The ``key(a)`` method accepts one argument with the array ``eltType`` and
+returns a value that supports all of comparison operations: ``<``, ``<=``,
+``>``, ``>=``, or ``==``.
```

Comment:
Actually I think it would be better if you said that the value returned by `key` must support all numeric operations, including comparison ( which you listed), and also things like - and <<.

I think it's more efficient to use - for our default comparator...
I'd use << and & for radix sorting...